watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Sites/SiteAbstractFragment.java

```diff
@@ -0,0 +1,217 @@
+                } catch (Exception e) {
+                    has_errors = true;
+                    mZipField.setError("Please enter a valid ZIP code");
+                }
+
+                if (has_errors) {
+                    return;
+                }
+
+                submitListener();
+
+                SiteListFragment returnFragment = SiteListFragment.newInstance();
+                mMainActivity.replaceFragment(returnFragment);
+            }
+        };
+    }
+
+    /**
+     * Creates and sends a request for either editing or creating a new site.
+     * sets default values for Lat, Long, TaskCount, and MiniSiteCount.
+     * @param type CREATE or EDIT.
+     * @param new_site The site to be editted, or null.
+     */
+    protected void createSite(String type, Site new_site) {
+        if (type.equals(CREATE)){
+            new_site = new Site();
+            new_site.setLatitude("0");
+            new_site.setLongitude("0");
+            new_site.setTasksCount(0);
+            new_site.setMiniSitesCount(0);
+        }
+
+        new_site.setName(mTitleField.getText().toString());
+        new_site.setDescription(mDescriptionField.getText().toString());
+        new_site.setStreet(mAddressField.getText().toString());
+        new_site.setCity(mCityField.getText().toString());
+        new_site.setState(mStateField.getText().toString());
+        new_site.setZipCode(Integer.valueOf(mZipField.getText().toString()));
+
+        createSiteRequest(type, new_site);
+    }
+
+    /**
+     * Sets Empty Error messages.
+     * @param field The field name (eg. City, Street...)
+     * @param editText some EditText
+     */
+    private void setEmpty(String field, EditText editText) { editText.setError(field + " can't be blank!"); }
+
+    public void createSiteRequest(String type, Site site){
+        HashMap<String, JSONObject> params = new HashMap<String, JSONObject>();
+
+        if (type.equals(CREATE)) {
+            CreateSiteRequest createSiteRequest = new CreateSiteRequest(mMainActivity, site, params, new Response.Listener<Site>() {
+                @Override
+                public void onResponse(Site site) {
+                    Log.e("successful site", "creation");
+                }
+            });
+            mNetworkManager.getRequestQueue().add(createSiteRequest);
+            SiteListFragment returnFragment = SiteListFragment.newInstance();
+            mMainActivity.replaceFragment(returnFragment);
+        }
+        else {
+            EditSiteRequest editSiteRequest = new EditSiteRequest(mMainActivity, site, params, new Response.Listener<Site>() {
+                @Override
+                public void onResponse(Site site) {
+                    Log.e("successful site", "edit");
+                }
+            });
+            mNetworkManager.getRequestQueue().add(editSiteRequest);
+            mMainActivity.getSupportFragmentManager().popBackStack();
```

Comment:
why do we use back stack to go back here and use a new instance of the fragment above?